diff -Nur yolov5/data/coco.yaml yolov5_gpu/data/coco.yaml
--- yolov5/data/coco.yaml	2022-07-15 06:48:34.811345247 +0000
+++ yolov5_gpu/data/coco.yaml	2022-07-15 06:45:17.711861275 +0000
@@ -8,7 +8,7 @@
 
 
 # Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
-path: ../datasets/coco  # dataset root dir
+path: /data  # dataset root dir
 train: train2017.txt  # train images (relative to 'path') 118287 images
 val: val2017.txt  # val images (relative to 'path') 5000 images
 test: test-dev2017.txt  # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794
diff -Nur yolov5/train.py yolov5_gpu/train.py
--- yolov5/train.py	2022-07-15 06:48:34.815345234 +0000
+++ yolov5_gpu/train.py	2022-07-15 06:46:57.671527539 +0000
@@ -150,7 +150,8 @@
     # Optimizer
     nbs = 64  # nominal batch size
     accumulate = max(round(nbs / batch_size), 1)  # accumulate loss before optimizing
-    hyp['weight_decay'] *= batch_size * accumulate / nbs  # scale weight_decay
+    # hyp['weight_decay'] *= batch_size * accumulate / nbs  # scale weight_decay
+    hyp['weight_decay'] = 0.001
     LOGGER.info(f"Scaled weight_decay = {hyp['weight_decay']}")
 
     g0, g1, g2 = [], [], []  # optimizer parameter groups
@@ -539,7 +540,7 @@
         train(opt.hyp, opt, device, callbacks)
         if WORLD_SIZE > 1 and RANK == 0:
             LOGGER.info('Destroying process group... ')
-            dist.destroy_process_group()
+            # dist.destroy_process_group()
 
     # Evolve hyperparameters (optional)
     else:
